server/src/main/java/org/eclipse/openvsx/security/DefaultAuthUser.java

```diff
@@ -0,0 +1,68 @@
+/********************************************************************************
+ * Copyright (c) 2023 Ericsson and others
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ ********************************************************************************/
+package org.eclipse.openvsx.security;
+
+import org.springframework.security.oauth2.core.user.OAuth2User;
+
+public class DefaultAuthUser implements AuthUser {
+
+    final String authId;
+    final String avatarUrl;
+    final String email;
+    final String fullName;
+    final String loginName;
+    final String providerId;
+    final String providerUrl;
+
+    public DefaultAuthUser(String providerId, OAuth2User oauth2User) {
+        authId = oauth2User.getName();
+        avatarUrl = oauth2User.getAttribute("avatar_url");
+        email = oauth2User.getAttribute("email");
+        fullName = oauth2User.getAttribute("name");
+        loginName = oauth2User.getAttribute("login");
```

Comment:
After looking closer at this. I wonder what's the difference between `loginName` and `authId` on GitHub. I suggest to name this class GitHubAuthUser.

Reply:
> After looking closer at this. I wonder what's the difference between `loginName` and `authId` on GitHub.

The naming of attributes like `oauth2User.getAttribute("login");` is completely arbitrary on the OAuth2 provider side. GitLab's `authId` might be GitHub's `login` equivalent just because.

> If the expected data is stored in different attributes it will be possible to bridge it by implementing the proper `AuthUser`.

This was my original idea with this PR: the code would no longer hardcode attribute names in several locations: Attribute access would be abstracted in one place: within `AuthUser` implementations.

> I wonder whether in a more generic environment one should use whatever is set via the `spring.security.oauth2.client.provider.[id[.user-name-attribute` configuration value instead of using the hardcoded name "login"

It's not a bad idea, I'll see what I can do about it later.

Reply:
Looked into it and I'd like to keep the PR as is for now.

My main issue is how rigid the code currently is: The `OAuth2UserServices` hardcodes the expect registration IDs such as `github` and `eclipse`. In our case we just override components to link to an alternative OAuth2 provider so we're skipping that problematic logic from OpenVSX. This `AuthUser` abstraction is enough for us to then propagate information down the rest of the system.

Feel free to contribute a more generic authentication system later.